### PR TITLE
chore(project setup): setup project CU-3w90xe9

### PR DESCRIPTION
**Pull request by kelue**
**What does this pull request do?**

       Setup folder structure, initializes database connection and install npm dependencies.

**Description of tasks completed**
- Sets up the project structure
- installs dependencies
- implement database connection
- setup database folder structure 

**How to test?**
- clone the repo
- install the dependencies by running `npm i`
- run `npm run dev` to start the server

### CODE REVIEW

**Comment by Endowmissy** [comment]
Task linked: [CU-3w90xe9 Project Setup](https://app.clickup.com/t/3w90xe9)

File: database/database.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+require('dotenv').config();
+
+const { DB_USER, DB_PASSWORD, DB, DB_HOST } = process.env;
+// import env variables for test environment
+const { TESTDB_USER, TESTDB_PASSWORD, TESTDB, TESTDB_HOST } = process.env;
```
Comment (Endowmissy) [inline database/database.js]: why destructure this seperately when you can do that on line 3

File: .env.example [hunk 2]
```diff
@@ -0,0 +1,4 @@
+DB="database name"
```
Comment (Endowmissy) [inline .env.example]: @kelue change DB to DB_NAME

File: src/assests/constants.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+module.exports = {
+  // HTTP status as constants to improve readability of code
+  SUCCESS: 200,
+  CREATED: 201,
+  BAD_REQUEST: 400,
+  UNAUTHORISED: 401,
+  FORBIDDEN: 403,
+  NOT_FOUND: 404,
+  ERROR: 500,
```
Comment (Endowmissy) [inline src/assests/constants.js]: @kelue change ERROR to SERVER_ERROR and also add ERROR to ERROR names e.g NOT_FOUND_ERROR

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+    "extends": ["airbnb","prettier"],
+    "plugins": ["prettier"],
+    "ignorePatterns": ["/tests/*.test.js"],
+    "rules": {
+      "prettier/prettier": ["error"]
+    }
+}
```
Comment (Endowmissy) [inline .eslintrc]: Add new line to the end of all your files

Comment (kelue) [reply .eslintrc]: That is already in the default prettier settings

File: src/server.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+require('dotenv').config();
+const express = require('express');
+const db = require('../database/config');
```
Comment (Endowmissy) [inline src/server.js]: since you are not reusing db, no need to declare a varaible, just require database config file

**Review by Endowmissy — APPROVED** [review]
@kelue Good work. Weldone.

**Review by bright258 — APPROVED** [review]

**Review by bright258 — APPROVED** [review]

**Pull request merged**